### PR TITLE
app/dcs/api/s3/s3-compatibility: update with the latest storj.io/gateway/docs/s3-compatibility.md

### PR DESCRIPTION
**Pull request by amwolff**
This updates the page with the latest storj.io/gateway/docs/s3-compatibility.md to make it the ground truth and prepare for removal of storj.io/gateway/docs/s3-compatibility.md. It also adds another context for GetBucketLocation from https://github.com/storj/docs/pull/285, superseding this pull request.

### CODE REVIEW

**Comment by amwolff** [comment]
It looks like the exact wording for the LocationConstraint section is still pending an approval, so better not to merge it without an explicit approval from, e.g., @onionjake.

**Comment by amwolff** [comment]
It looks like the wording for the LocationConstraint section is still pending approval, so better not to merge this without explicit approval from, e.g., @onionjake.

**Comment by halkyon** [comment]
Do you also want to prepare the change to remove the docs in the gateway repo, so we don't have them in two places any longer?

File: app/dcs/api/s3/s3-compatibility/page.md [hunk 1]
```diff
@@ -11,104 +11,104 @@ The Storj S3-compatible Gateway supports a RESTful API that is compatible with t
...
+| ListBucketInventoryConfigurations           | No      | No                                                           |                                                     |
+| ListBucketMetricsConfigurations             | No      | No                                                           |                                                     |
+| ListBuckets                                 | Full    |                                                              |                                                     |
+| ListMultipartUploads                        | Partial | Planned full. See https://github.com/storj/roadmap/issues/20 | See ListMultipartUploads section                    |
+| ListObjectVersions                          | No      | Planned. See https://github.com/storj/roadmap/issues/23      |                                                     |
+| ListObjects                                 | Partial | Planned full. See https://github.com/storj/roadmap/issues/20 | See ListObjects section                             |
```
Comment (ferristocrat) [inline app/dcs/api/s3/s3-compatibility/page.md:77]:
```suggestion
| ListObjects                                 | Partial | We may support lexicographical ordering in the future.  | See ListObjects section                             |
```

Comment (ferristocrat) [reply app/dcs/api/s3/s3-compatibility/page.md:77]: I think we should get rid of references to https://github.com/storj/roadmap/issues/20

File: app/dcs/api/s3/s3-compatibility/page.md [hunk 2]
```diff
@@ -11,104 +11,104 @@ The Storj S3-compatible Gateway supports a RESTful API that is compatible with t
...
+| ListObjectsV2                               | Partial | Planned full. See https://github.com/storj/roadmap/issues/20 | See ListObjects section                             |
```
Comment (ferristocrat) [inline app/dcs/api/s3/s3-compatibility/page.md:78]:
```suggestion
| ListObjectsV2                               | Partial | We may support lexicographical ordering in the future. | See ListObjects section                             |
```

File: app/dcs/api/s3/s3-compatibility/page.md [hunk 3]
```diff
@@ -11,104 +11,104 @@ The Storj S3-compatible Gateway supports a RESTful API that is compatible with t
...
+| ListBucketAnalyticsConfigurations           | No      | No                                                           |                                                     |
+| ListBucketIntelligentTieringConfigurations  | No      | No                                                           |                                                     |
+| ListBucketInventoryConfigurations           | No      | No                                                           |                                                     |
+| ListBucketMetricsConfigurations             | No      | No                                                           |                                                     |
+| ListBuckets                                 | Full    |                                                              |                                                     |
+| ListMultipartUploads                        | Partial | Planned full. See https://github.com/storj/roadmap/issues/20 | See ListMultipartUploads section                    |
```
Comment (ferristocrat) [inline app/dcs/api/s3/s3-compatibility/page.md:75]:
```suggestion
| ListMultipartUploads                        | Partial | We may support lexicographical ordering in the future. | See ListMultipartUploads section                    |
```

File: app/dcs/api/s3/s3-compatibility/page.md [hunk 4]
```diff
@@ -11,104 +11,104 @@ The Storj S3-compatible Gateway supports a RESTful API that is compatible with t
...
+| PutObjectTagging                            | Full    |                                                              | Tags can be modified outside of tagging endpoints   |
+| PutPublicAccessBlock                        | No      | No                                                           |                                                     |
+| RestoreObject                               | No      | No                                                           |                                                     |
+| SelectObjectContent                         | No      | No                                                           |                                                     |
+| UploadPart                                  | Full    |                                                              |                                                     |
+| UploadPartCopy                              | No      | Planned. See https://github.com/storj/roadmap/issues/40      |                                                     |
```
Comment (ferristocrat) [inline app/dcs/api/s3/s3-compatibility/page.md:110]:
```suggestion
| UploadPartCopy                              | No      | Partial Support Planned. See https://github.com/storj/roadmap/issues/76      |                                                     |
```

File: app/dcs/api/s3/s3-compatibility/page.md [hunk 5]
```diff
@@ -120,6 +120,19 @@ Full compatibility means that we support all features of a specific action
...
+
+This is currently supported in Gateway-MT only.
+
+#### LocationConstraint
+
+Specifies the placement region where the bucket data is stored:
+
+|     Value     |                                                        Description                                                        |
+|:-------------:|:-------------------------------------------------------------------------------------------------------------------------:|
+| `global`      | Stored on the entire globe connected to the public network                                                                |
```
Comment (ferristocrat) [inline app/dcs/api/s3/s3-compatibility/page.md:133]:
```suggestion
| `global`      | Stored on nodes connected to the global public network                                                                |
```

Comment (ferristocrat) [reply app/dcs/api/s3/s3-compatibility/page.md:133]: It read a little bit weird. I think this is better, but I'm not SUPER opinionated here.

File: app/dcs/api/s3/s3-compatibility/page.md [hunk 6]
```diff
@@ -120,6 +120,19 @@ Full compatibility means that we support all features of a specific action
 except for features that rely on other actions that we haven't fully
 implemented.
 
+### GetBucketLocation
+
+This is currently supported in Gateway-MT only.
+
+#### LocationConstraint
+
+Specifies the placement region where the bucket data is stored:
```
Comment (ferristocrat) [inline app/dcs/api/s3/s3-compatibility/page.md:129]:
```suggestion
Specifies the placement region where the object data for a bucket is stored:
```

Comment (ferristocrat) [reply app/dcs/api/s3/s3-compatibility/page.md:129]: Should we specific OBJECT data? Given that this is grabbed from the satellite, this is more accurate right?

**Review by ferristocrat — CHANGES_REQUESTED** [review]

**Review by onionjake — APPROVED** [review]

**Comment by amwolff** [comment]
I pushed 7400cbd, which fixes the formatting of the tables and consistency of punctuation and capital letters usage after Michael's additions.

**Comment by onionjake** [comment]
This is good to go as far as I'm concerned. @ferristocrat I think you need to mark it approved because you reviewed with comments?

**Review by ferristocrat — APPROVED** [review]

**Pull request merged**